Stripe/STPCardParams.swift

```diff
@@ -0,0 +1,195 @@
+    get {
+      return address.state
+    }
+    set(addressState) {
+      address.state = addressState
+    }
+  }
+  /// The zip code of the cardholder's address
+
+  @objc var addressZip: String? {
+    get {
+      return address.postalCode
+    }
+    set(addressZip) {
+      address.postalCode = addressZip
+    }
+  }
+  /// The country of the cardholder's address
+
+  @objc public var addressCountry: String? {
+    get {
+      return address.country
+    }
+    set(addressCountry) {
+      address.country = addressCountry
+    }
+  }
+
+  override init() {
+    address = STPAddress()
+    super.init()
+    additionalAPIParameters = [:]
+  }
+
+  // MARK: - Description
+  /// :nodoc:
+  @objc public override var description: String {
+    let props = [
+      // Object
+      String(format: "%@: %p", NSStringFromClass(STPCardParams.self), self),
+      // Basic card details
+      "last4 = \(last4() ?? "")",
+      String(format: "expMonth = %lu", UInt(expMonth)),
+      String(format: "expYear = %lu", UInt(expYear)),
+      "cvc = \(((cvc) != nil ? "<redacted>" : nil) ?? "")",
+      // Additional card details (alphabetical)
+      "currency = \(currency ?? "")",
+      // Cardholder details
+      "name = \(((name) != nil ? "<redacted>" : nil) ?? "")",
+      "address = <redcated>",
+    ]
+
+    return "<\(props.joined(separator: "; "))>"
+  }
+
+  // MARK: - STPFormEncodable
+  class func rootObjectName() -> String? {
+    return "card"
+  }
+
+  class func propertyNamesToFormFieldNamesMapping() -> [String: String] {
+    return [
+      NSStringFromSelector(#selector(getter:number)): "number",
+      NSStringFromSelector(#selector(getter:cvc)): "cvc",
+      NSStringFromSelector(#selector(getter:name)): "name",
+      NSStringFromSelector(#selector(getter:addressLine1)): "address_line1",
+      NSStringFromSelector(#selector(getter:addressLine2)): "address_line2",
+      NSStringFromSelector(#selector(getter:addressCity)): "address_city",
+      NSStringFromSelector(#selector(getter:addressState)): "address_state",
+      NSStringFromSelector(#selector(getter:addressZip)): "address_zip",
+      NSStringFromSelector(#selector(getter:addressCountry)): "address_country",
+      NSStringFromSelector(#selector(getter:expMonth)): "exp_month",
+      NSStringFromSelector(#selector(getter:expYear)): "exp_year",
+      NSStringFromSelector(#selector(getter:currency)): "currency",
+    ]
+  }
+
+  // MARK: - Deprecated methods
+
+  // MARK: - NSCopying
+  /// :nodoc:
+  @objc
+  public func copy(with zone: NSZone? = nil) -> Any {
+    let copyCardParams = STPCardParams()
+
+    copyCardParams.number = number
+    copyCardParams.expMonth = expMonth
+    copyCardParams.expYear = expYear
+    copyCardParams.cvc = cvc
+
+    // Use ivar to avoid setName:/setAddress: behavior that'd possibly overwrite name/address.name
+    copyCardParams.name = name
+    copyCardParams.address = address.copy() as? STPAddress ?? address
```

Comment:
this should probably bet `?? STPAddress()`?

Reply:
Let's do `as! STPAddress`, we have weirder problems going on if `copy` gives us something that isn't castable to an STPAddress